test/v3/addLiquidity/addLiquidity.liquiditybootstrapping.integration.test.ts

```diff
@@ -0,0 +1,253 @@
+                ],
+            },
+            permit2,
+        );
+
+        const txOutput = await sendTransactionGetBalances(
+            [WETH.address, BAL.address],
+            client,
+            testAddress,
+            initPoolBuildOutput.to,
+            initPoolBuildOutput.callData,
+            initPoolBuildOutput.value,
+        );
+        assertInitPool(initPoolInput, { txOutput, initPoolBuildOutput });
+    });
+    test('add liquidity', async () => {
+        const addLiquidity = new AddLiquidityV3();
+        const input: AddLiquidityInput = {
+            chainId: chainId,
+            rpcUrl: rpcUrl,
+            sender: (await client.getAddresses())[0],
+            userData: '0x',
+            amountsIn: [
+                {
+                    address: WETH.address,
+                    rawAmount: parseUnits('2', WETH.decimals),
+                    decimals: WETH.decimals,
+                },
+                {
+                    address: BAL.address,
+                    rawAmount: parseUnits('33', BAL.decimals),
+                    decimals: BAL.decimals,
+                },
+            ],
+            kind: AddLiquidityKind.Unbalanced,
+        };
+
+        const poolState: PoolState = {
+            id: poolAddress,
+            address: poolAddress,
+            type: 'LIQUIDITYBOOTSTRAPPING',
+            tokens: [
+                {
+                    address: WETH.address,
+                    index: 0,
+                    decimals: WETH.decimals,
+                },
+                {
+                    address: BAL.address,
+                    index: 1,
+                    decimals: BAL.decimals,
+                },
+            ],
+            protocolVersion: 3,
+        };
+
+        const addLiquidityQueryOutput = await addLiquidity.query(
+            input,
+            poolState,
+        );
+
+        // generate the build call input
+        const addLiquidityBuildInput = {
+            ...addLiquidityQueryOutput,
+            slippage: Slippage.fromPercentage('1'),
+            wethIsEth: false,
+            userData: '0x' as Hex,
+        };
+
+        const permit2 = await Permit2Helper.signAddLiquidityApproval({
+            ...addLiquidityBuildInput,
+            client,
+            owner: testAddress,
+        });
+
+        const addLiquidityBuildCallOutput = addLiquidity.buildCallWithPermit2(
+            addLiquidityBuildInput,
+            permit2,
+        );
+
+        const txOutput = await sendTransactionGetBalances(
+            [WETH.address, BAL.address, poolAddress],
+            client,
+            testAddress,
+            addLiquidityBuildCallOutput.to,
+            addLiquidityBuildCallOutput.callData,
+            addLiquidityBuildCallOutput.value,
+        );
+
+        // tokens leaver wallet, bpt enter wallet (all deltas are positive)
+        txOutput.balanceDeltas.forEach((balance) => {
+            expect(balance).toBeGreaterThan(0n);
+        });
```

Comment:
I'm leaning towards not having these tests at all, but if we do keep them, you should compare query with actual transaction results. Simply checking if all is positive isn't enough.
There are helpers to assert add liquidity operations you could reuse.

Reply:
I have removed the tests